src/main/java/org/zalando/nakadi/partitioning/HashPartitionStrategy.java

```diff
@@ -34,15 +34,16 @@ public String calculatePartition(final EventType eventType, final JSONObject eve
                     // The problem is that JSONObject doesn't override hashCode(). Therefore convert it to
                     // a string first and then use hashCode()
                     .map(pkf -> EventCategory.DATA.equals(eventType.getCategory()) ? DATA_PATH_PREFIX + pkf : pkf)
-                    .map(wrapFunction(okf -> traversableJsonEvent.get(okf).toString().hashCode()))
+                    .map(Try.wrap(okf -> traversableJsonEvent.get(okf).toString().hashCode()))
```

Comment:
Just in few words, what is the advantage of switching to this Try instead of exception wrapper?
Currently I see one disadvantage that the code became more verbose as we always call `Try::getOrThrow`.